modules/tensor_mechanics/src/materials/ADComputeIncrementalShellStrain.C

```diff
@@ -0,0 +1,544 @@
+
+        // corresponding to strain(1,1)
+        (*_B[j])[i](1, k) = _dphideta_map[k][i] * (*_dxyz_deta[j])[i](0);
+        (*_B[j])[i](1, 4 + k) = _dphideta_map[k][i] * (*_dxyz_deta[j])[i](1);
+        (*_B[j])[i](1, 8 + k) = _dphideta_map[k][i] * (*_dxyz_deta[j])[i](2);
+        (*_B[j])[i](1, 12 + k) = _dphideta_map[k][i] * _t_points[j](0) / 2.0 * _thickness[i] *
+                                 (-_V2[k] * (*_dxyz_deta[j])[i]);
+        (*_B[j])[i](1, 16 + k) = _dphideta_map[k][i] * _t_points[j](0) / 2.0 * _thickness[i] *
+                                 (_V1[k] * (*_dxyz_deta[j])[i]);
+
+        // corresponding to strain(2,2) = 0
+
+        // corresponding to strain(0,1)
+        (*_B[j])[i](2, k) = 0.5 * (_dphideta_map[k][i] * (*_dxyz_dxi[j])[i](0) +
+                                   _dphidxi_map[k][i] * (*_dxyz_deta[j])[i](0));
+        (*_B[j])[i](2, 4 + k) = 0.5 * (_dphideta_map[k][i] * (*_dxyz_dxi[j])[i](1) +
+                                       _dphidxi_map[k][i] * (*_dxyz_deta[j])[i](1));
+        (*_B[j])[i](2, 8 + k) = 0.5 * (_dphideta_map[k][i] * (*_dxyz_dxi[j])[i](2) +
+                                       _dphidxi_map[k][i] * (*_dxyz_deta[j])[i](2));
+        (*_B[j])[i](2, 12 + k) =
+            0.25 * _t_points[j](0) * _thickness[i] * -_V2[k] *
+            (_dphideta_map[k][i] * (*_dxyz_dxi[j])[i] + _dphidxi_map[k][i] * (*_dxyz_deta[j])[i]);
+        (*_B[j])[i](2, 16 + k) =
+            0.25 * _t_points[j](0) * _thickness[i] * _V1[k] *
+            ((*_dxyz_deta[j])[i] * _dphidxi_map[k][i] + (*_dxyz_dxi[j])[i] * _dphideta_map[k][i]);
+      }
+
+      _g3_A = _thickness[i] / 4.0 * (_node_normal[2] + _node_normal[3]);
+      _g3_C = _thickness[i] / 4.0 * (_node_normal[0] + _node_normal[1]);
+      _g3_B = _thickness[i] / 4.0 * (_node_normal[0] + _node_normal[3]);
+      _g3_D = _thickness[i] / 4.0 * (_node_normal[1] + _node_normal[2]);
+
+      _g1_A = 0.5 * ((*_nodes[2]) - (*_nodes[3])) +
+              _t_points[j](0) / 4.0 * _thickness[i] * (_node_normal[2] - _node_normal[3]);
+      _g1_C = 0.5 * ((*_nodes[1]) - (*_nodes[0])) +
+              _t_points[j](0) / 4.0 * _thickness[i] * (_node_normal[1] - _node_normal[0]);
+      _g2_B = 0.5 * ((*_nodes[3]) - (*_nodes[0])) +
+              _t_points[j](0) / 4.0 * _thickness[i] * (_node_normal[3] - _node_normal[0]);
+      _g2_D = 0.5 * ((*_nodes[2]) - (*_nodes[1])) +
+              _t_points[j](0) / 4.0 * _thickness[i] * (_node_normal[2] - _node_normal[1]);
+
+      updateGVectors(); // for large strain problems
+
+      // corresponding to strain(0,2)
+      for (unsigned int component = 0; component < 3; component++)
+      {
+        (*_B[j])[i](3, 2 + component * 4) = 1.0 / 8.0 * (1.0 + _2d_points[i](1)) * _g3_A(component);
+        (*_B[j])[i](3, 3 + component * 4) =
+            1.0 / 8.0 * (1.0 + _2d_points[i](1)) * -_g3_A(component);
+        (*_B[j])[i](3, 1 + component * 4) = 1.0 / 8.0 * (1.0 - _2d_points[i](1)) * _g3_C(component);
```

Comment:
replace the instances of 1/8 with 0.125.  Less operations and still self-explanatory.